src/tutorials/pet-shop.md

```diff
@@ -374,6 +376,100 @@ Since arrays can only return a single value given a single key, we create our ow
    ```
 
 Note the **memory** attribute on `adopters`. The memory attribute tells Solidity to temporarily store the value in memory, rather than saving it to the contract's storage. Since `adopters` is an array, and we know from the first adoption test that we adopted pet `expectedPetId`, we compare the testing contracts address with location `expectedPetId` in the array.
+</details>
+
+## Testing the smart contract using Javascript 
+
+<details>
+<summary> Expand This Section </summary>
+Truffle is very flexible when it comes to smart contract testing, in that tests can be written either in JavaScript or Solidity. In this tutorial, we'll be writing our tests in Javascript using chai and mocha library.<br/>
+
+1. Create a new file named `testAdoption.test.js` in the `test/` directory.<br/>
+2. Add the following content to the `testAdoption.test.js` file:<br/>
+
+  ```
+  const Adoption = require("../contracts/Adoption.sol");
+  
+  contract("Adoption", (accounts) => {
+    let adoption;
+    let expectedPetId;
+    
+    before(async() => {
+        adoption = await Adoption.deployed();
+    });
+
+    describe("adopting a pet and retrieving account addresses", async() => {
+      before("adopt a pet using accounts[0]", async () => {
+      await adoption.adopt(8, { from: accounts[0] });
+      expectedAdopter = accounts[0];
+    });
+    });
+  });
+
+  ```
+  We start the contract by importing : 
+  * ```Adoption.sol```:The smart contract we want to test
```

Comment:
We should remove the note about importing chai. Later we could tell the reader that Truffle automatically imports it for you and that is why we can use `assert`. So this probably should not be a list any more if we remove the line about chai.

So I think it should be "We begin our test by importing our `Adoption` contract using `artifacts.require`."